[nmstate-1.3] ip: Preserve the IP address order when applying

cathay4t opened the pull request:
When applying the IP address, we should preserve the order for use case
whether user is expecting non-first ones been set with `secondary` flag.

In RHEL/CentOS 8, NetworkManager is using reverted IPv6 address
according to
https://bugzilla.redhat.com/show_bug.cgi?id=2139443

Hence downstream nmstate will ship additional patch to fix it.
The upstream nmstate will not revert the IPv6 address list before
sending to NM.

The downstream build of RHEL 8 has different behaviour than copr build
from git main branch. It is hard to tell whether we are using downstream
build or git build at runtime, hence we ship the
`test_preserve_ipv6_addresses_order` test in RHEL 8.

ffmancera reviewed (changes requested):
The changes are not correct, the test is failing, could you check it out? Thanks!

cathay4t committed:
ip: Preserve the IP address order when applying

When applying the IP address, we should preserve the order for use case
whether user is expecting non-first ones been set with `secondary` flag.

In RHEL/CentOS 8, NetworkManager is using reverted IPv6 address
according to
https://bugzilla.redhat.com/show_bug.cgi?id=2139443

Hence downstream nmstate will ship additional patch to fix it.
The upstream nmstate will not revert the IPv6 address list before
sending to NM.

The downstream build of RHEL 8 has different behaviour than copr build
from git main branch. It is hard to tell whether we are using downstream
build or git build at runtime, hence we ship the
`test_preserve_ipv6_addresses_order` test in RHEL 8.

Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
